drivers/SmartThings/matter-appliance/src/matter-extractor-hood/init.lua

```diff
@@ -104,10 +160,11 @@ local function fan_mode_sequence_handler(driver, device, ib, response)
 end
 
 local function fan_speed_percent_attr_handler(driver, device, ib, response)
-  local speed = ib.data.value
-  if speed ~= nil and speed <= 100 then
-    device:emit_event_for_endpoint(ib.endpoint_id, capabilities.fanSpeedPercent.percent(speed))
+  local speed = 0
+  if ib.data.value ~= nil then
+    speed = utils.clamp_value(ib.data.value, MIN_ALLOWED_PERCENT_VALUE, MAX_ALLOWED_PERCENT_VALUE)
   end
+  device:emit_event_for_endpoint(ib.endpoint_id, capabilities.fanSpeedPercent.percent(speed))
```

Comment:
we want to emit a 0 event if the input is null?

Reply:
Good point! I refactored this handler to use an early return if the input is null